pkg/platform/runtime/artifactcache/artifactcache.go

```diff
@@ -106,6 +106,20 @@ func (cache *ArtifactCache) Store(a artifact.ArtifactID, archivePath string) err
 	cache.mutex.Lock()
 	defer cache.mutex.Unlock()
 
+	// Replace an existing artifact in the cache.
+	// This would really only happen if a checksum validation fails and the artifact is silently
+	// re-downloaded from the platform.
+	if existingArtifact, found := cache.artifacts[a]; found {
+		path := existingArtifact.ArchivePath
+		logging.Debug("Replacing cached artifact '%s'", path)
+		err := os.Remove(path)
+		if err != nil {
+			return errs.Wrap(err, "Unable to overwrite existing artifact '%s'", path)
+		}
+		delete(cache.artifacts, existingArtifact.Id)
+		cache.currentSize -= existingArtifact.Size
+	}
```

Comment:
I think if checksum validation failed it should error out and not reach this point. But pending Pete's call here: https://activestate.slack.com/archives/CFN2A6P9N/p1658859008105439

Reply:
I can edit the comment to be more explicit about this happening if the cached artifact's checksum fails (e.g. replaced by a bad actor). If the validation fails for a downloaded artifact, the entire process will error out.

Reply:
Pete says this behavior as programmed is good (hit the cache first, then silently redownload from Platform if checksum fails; if Platform artifact checksum validation fails, simply error out reporting that).